doc/tests/empty.rst

```diff
@@ -1,11 +1,20 @@
 ``empty``
 =========
 
-``empty`` checks if a variable is empty:
+``empty`` checks if a variable is
+ - an empty string
+ - an empty array
+ - exactly ``false``
+ - exactly ``null``
+ - an instance of Countable_ and has a count_ of zero
```

Comment:
I tried to find a list of what in PHP built-ins implements Countable to be more specific, but I couldn't find a definitive list. It's also worth noting that even though PHP `count` works with a PHP `array`, the way the code in Twig is written it will not catch an empty array with that code, it later checks specifically for equality with an empty array.

I don't mind changing this, but I'm unsure of the correct phrasing that would be useful for Twig users.